backend/src/main/java/org/eclipse/tractusx/puris/backend/common/edc/logic/util/EDCRequestBodyBuilder.java

```diff
@@ -2,214 +2,211 @@
+        objectNode.put("@type", "CatalogRequest");
+        objectNode.put("counterPartyAddress", counterPartyDspUrl);
+        if(filter != null && !filter.isEmpty()) {
+            var querySpecNode = MAPPER.createObjectNode();
+            objectNode.set("querySpec", querySpecNode);
+            for (var entry : filter.entrySet()) {
+                querySpecNode.put(entry.getKey(), entry.getValue());
+            }
+        }
+        return objectNode;
+    }
+
     /**
-     * Build an EDC request body for the creation of an asset (using an order).
-     *
-     * @param orderUrl url where the published order can be received by the controlplane.
-     * @param orderId  id of the created asset (currently has to match policy and contract id).
-     * @return JsonNode used as requestBody for asset creation.
+     * Build a request body for a request to register an api method as an asset in DSP protocol.
+     * @param apiMethod - the API method you want to register
+     * @return - the request body
      */
-    public JsonNode buildAssetRequestBody(String orderUrl, String orderId) {
-        var assetRequest = MAPPER.createObjectNode();
-        var assetNode = MAPPER.createObjectNode();
-        var assetPropNode = MAPPER.createObjectNode();
-        assetNode.set("properties", assetPropNode);
-        assetPropNode.put("asset:prop:id", orderId);
-        assetPropNode.put("asset:prop:description", "EDC Demo Asset");
-        assetRequest.set("asset", assetNode);
-        var dataAddressNode = MAPPER.createObjectNode();
-        var propertiesNode = MAPPER.createObjectNode();
-        dataAddressNode.set("properties", propertiesNode);
-        propertiesNode.put("type", "HttpData");
-        propertiesNode.put("baseUrl", orderUrl);
-        assetRequest.set("dataAddress", dataAddressNode);
-        return assetRequest;
+    public JsonNode buildDSPCreateAssetBody(DT_ApiMethodEnum apiMethod) {
+        var body = MAPPER.createObjectNode();
+        var context = MAPPER.createObjectNode();
+        context.put("edc", "https://w3id.org/edc/v0.0.1/ns/");
+        context.put("cx-taxo", "https://w3id.org/catenax/taxonomy#");
+        context.put("cx-common", "https://w3id.org/catenax/ontology/common#");
+        context.put("dct", "https://purl.org/dc/terms/");
+        body.set("@context", context);
+
+        String apiId = variablesService.getApiAssetId(apiMethod);
+        body.put("@id", apiId);
+        var properties = MAPPER.createObjectNode();
+        properties.put("asset:prop:type", "api");
+        properties.put("asset:prop:apibusinessobject", "product-stock");
+        properties.put("asset:prop:version", variablesService.getPurisApiVersion());
+        properties.put("asset:prop:apipurpose", apiMethod.PURPOSE);
+        body.set("properties", properties);
+        var edcProperties = MAPPER.createObjectNode();
+        edcProperties.put("https://w3id.org/catenax/ontology/common#version", "2.0.0");
```

Comment:
I have created final constants. But I am not quite sure what you mean there with json ld

Reply:
when adding the namespace + property instead of having the short-notation + property this is called expansion and it's related to json-ld. Therefore I just would put the key String.format. But constant and common concatenation with + is also key. key aspect is the constant :)